src/components/EmailValidationForm.tsx

```diff
@@ -1,71 +1,88 @@
-}
-
-export interface EmailValidationFormDispatchProps {
-  validateEmail: (data: FormData) => Promise<void>;
 }
 
 export interface EmailValidationFormStateProps {
   error?: FetchErrorData;
 }
 
+export interface EmailValidationFormDispatchProps {
+  validateEmail: (data: FormData) => Promise<void>;
+}
+
 export interface EmailValidationFormProps extends EmailValidationFormOwnProps, EmailValidationFormDispatchProps, EmailValidationFormStateProps {}
 
 export class EmailValidationForm extends React.Component<EmailValidationFormProps, EmailValidationFormState> {
-
   constructor(props) {
     super(props);
-    this.sendEmail = this.sendEmail.bind(this);
-    this.state = { sent: false };
+    this.validate = this.validate.bind(this);
+    this.renderInfoText = this.renderInfoText.bind(this);
+    this.state = { validated: false };
   }
 
-  async sendEmail(data: FormData): Promise<void> {
-    await this.props.validateEmail(data);
-    await this.props.fetchLibrary(this.props.library.uuid);
-    this.setState({ sent: !this.props.error });
+  convert(email: string): string {
+    return (email.includes("_") ? email.replace("_", " ") : email.replace(" ", "_"));
   }
 
-  render(): JSX.Element {
-    let icon = this.state.sent ? <CheckSoloIcon /> : null;
-    let alreadyValidated = this.props.library.urls_and_contact.validated !== "Not validated";
-    let hasEmail = !!this.props.library.urls_and_contact.contact_email;
+  renderTitle(emailString: string, emailAddress: string, alreadyValidated: boolean): JSX.Element[] {
+    let titleString = `${emailString}${emailAddress ? ": " + emailAddress : ""}`;
+    let icon = ((alreadyValidated || this.state.validated) && !this.props.error) ? <CheckSoloIcon /> : <XIcon />;
+    return [<span>{titleString}</span>, icon];
+  }
+
+  async validate(email: string, data: FormData): Promise<void> {
+    data.append("email", email);
+    await this.props.validateEmail(data);
+    this.setState({ validated: true });
+  }
 
-    let buttonText = hasEmail ? "Validate email address" : "No email address configured";
-    let buttonContent = <span>{buttonText}{icon}</span>;
-    let infoText = (alreadyValidated && !this.state.sent && !this.props.error) ? "Already validated" : null;
-    let successText = `Successfully validated ${this.props.library.urls_and_contact.contact_email}`;
+  renderInfoText(emailString: string, emailAddress: string, validated: string, alreadyValidated: boolean): string {
+    if (!emailAddress) {
+      return `No ${emailString} configured`;
+    } else if (!alreadyValidated) {
+      return "Not validated";
+    } else if (!this.state.validated && !this.props.error) {
+      return `Validated: ${validated}`;
+    }
+  }
 
+  render(): JSX.Element {
+    let emailString = this.convert(this.props.email);
+    let emailAddress = this.props.libraryInfo[this.props.email];
+    let validated = this.props.libraryInfo[`${this.props.email.split("_")[0]}_validated`];
```

Comment:
I'm guessing that the `libraryInfo` object has a property like `contact_validated` which is what you are getting here? Some documentation on that would be helpful since it's not easily readable or easy to tell why you're doing this.

Reply:
@adriana-alter here too.

Reply:
I did; github is just being weird.
<img width="1006" alt="Screen Shot 2020-03-09 at 5 01 56 PM" src="https://user-images.githubusercontent.com/42178216/76257235-b77eab80-6227-11ea-96f7-87d27d453362.png">